interface/wx/print.h

```diff
@@ -705,23 +705,39 @@ class wxPrintout : public wxObject
         and maximum page values that the user can select, and the required page range to
         be printed.
 
-        If the user chose to print only selected pages in the MSW printing
-        dialog, then @a pageFrom and @a pageTo are used to limit the page range
-        and IsPageSelected() is called later to query whether the page is
-        selected and so should be printed.
-
-        If the user chose to print the current page, then @a pageFrom and
-        @a pageTo should be both set to the current page number.
-
         By default this returns (1, 32000) for the page minimum and maximum values, and
         (1, 1) for the required page range.
 
         @a minPage must be greater than zero and @a maxPage must be greater
         than @a minPage.
+
+        @deprecated
+        The newer method that support multiple ranges should be used instead.
     */
     virtual void GetPageInfo(int* minPage, int* maxPage, int* pageFrom,
                              int* pageTo);
 
+    /**
+        Called by the framework to obtain information from the application about minimum
+        and maximum page values that the user can select, and the required page ranges to
+        be printed.
+
+        If the user chose to print only selected pages or the current page in the MSW printing
+        dialog, then @a pageRanges are used to specify the pages to print.
+
+        If the user chose to print the current page, then @a pageRanges should contain one range
+        and the range values are both set to the current page number.
+
+        By default this returns (1, 32000) for the page minimum and maximum values, and
+        (1, 1) for the required page range.
+
+        @a minPage must be greater than zero and @a maxPage must be greater
+        than @a minPage.
+
+        @since 3.3.0.
+    */
+    virtual void GetPageInfo(int* minPage, int* maxPage, wxVector<wxPrintPageRange>* pageRanges);
```

Comment:
The function name is wrong here:
```suggestion
    virtual void GetPageInfoRanges(int* minPage, int* maxPage, wxVector<wxPrintPageRange>* pageRanges);
```